### PR TITLE
Build polyfills inside Docker image instead of copying from the host.

### PR DESCRIPTION
**Pull request by JakeChampion**
This will avoid the situation whereby polyfills on the host machine are used in the deployed Docker image. 

TL;DR:
Running this command to build the image and run it:
`rm -rf node_modules && rm -rf polyfills/__dist && npm run container-build && npm run container-test`
Results:
<details>
<summary>

Running on master: fails</summary>

```
>...<Lots of installation messages>...

Caught exception Error: No polyfill sources found.  Run `grunt build` to build them
    at Error (native)
    at Object.<anonymous> (/app/lib/sources.js:25:8)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/lib/index.js:8:20)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/service/routes/api.js:5:20)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/service/index.js:46:9)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Module.require (module.js:466:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/app/bin/polyfill-service:8:20)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Function.Module.runMain (module.js:575:10)
    at startup (node.js:159:18)
    at node.js:444:3
```

</details>

<details>
<summary>

Running on this PR: works</summary>

```
>...<Lots of installation messages>...
Found 862 polyfills
Graphite reporting is disabled.  To enable, set GRAPHITE_HOST
polyfill-service@3.8.1 started pid=16 port=3000 env=production
```

</details>

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -8,4 +8,8 @@ Dockerfile
 fastly-config.vcl
 LICENSE.md
 Makefile
-README.md
\ No newline at end of file
+README.md
+/polyfills/__dist/
+.env
+.service.pid
+npm-debug.log
```
Comment (triblondon) [inline .dockerignore:15]: Should we sort this list?

File: Gruntfile.js [hunk 2]
```diff
@@ -90,10 +91,16 @@ module.exports = function(grunt) {
...
+	if (process.env.NODE_ENV === 'production') {
+		require('./tasks/buildsources')(grunt);
+		require('./tasks/updatelibrary')(grunt);
+		grunt.loadNpmTasks('grunt-contrib-clean');
+	} else {
+		grunt.loadTasks('tasks');
+		grunt.loadNpmTasks('grunt-contrib-clean');
+		grunt.loadNpmTasks('grunt-contrib-watch');
+		grunt.loadNpmTasks('grunt-simple-mocha');
+	}
```
Comment (triblondon) [inline Gruntfile.js:103]: Not sure I follow the reason for this fork?

File: package.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
   },
   "scripts": {
     "start": "bin/polyfill-service",
-    "prepublish": "grunt updatelibrary && grunt build",
+    "prepublish": "npm run build",
+    "build": "grunt updatelibrary && grunt build",
```
Comment (triblondon) [inline package.json]: Can we make this a grunt task?  It seems like making a confusing situation more confusing to define this task in package scripts.

File: package.json [hunk 4]
```diff
@@ -36,13 +37,18 @@
...
     "blocked": "^1.1.0",
     "denodeify": "^1.2.1",
     "dotenv": "^1.2.0",
     "exists-sync": "0.0.3",
     "expect.js": "^0.3.1",
     "express": "^4.4.4",
     "graphite": "0.0.6",
+		"grunt": "~0.4.0",
+		"grunt-cli": "~0.1.6",
+		"grunt-contrib-clean": "^0.6.0",
```
Comment (triblondon) [inline package.json]: Are there indent issues here?

File: package.json [hunk 5]
```diff
@@ -62,14 +68,9 @@
...
-    "babel-preset-es2015": "^6.1.18",
     "batch": "^0.5.1",
     "fastly": "Financial-Times/fastly#v2.5.0",
-    "grunt": "~0.4.0",
-    "grunt-cli": "~0.1.6",
-    "grunt-contrib-clean": "^0.6.0",
-    "grunt-contrib-watch": "^0.6.1",
+		"grunt-contrib-watch": "^0.6.1",
     "grunt-simple-mocha": "^0.4.0",
     "intl": "^1.2.4",
```
Comment (triblondon) [inline package.json]: Some polyfills in devdeps, some in deps.  Should be the same?

File: Gruntfile.js (hunk 2 above)
Comment (JakeChampion) [inline Gruntfile.js:103]: In an attempt to reduce the amount of dependencies installed inside the Docker image, the `buildsources`,`updatelibrary` and `clean` tasks are required to install and build our polyfills. All the other grunt tasks are only required for development tasks.

**Comment by JakeChampion** [comment]
@triblondon Updated the PR based on feedback given. What do you think about https://github.com/Financial-Times/polyfill-service/pull/770#discussion_r72587903 ?

**Comment by triblondon** [comment]
Do we need both the `build` and `installpolyfills` tasks?  Feel free to consolidate if possible.  Otherwise 👍

**Pull request merged**